src/test/java/com/wb/WeatherForecastApiApplicationUnitTests.java

```diff
@@ -1,85 +1,54 @@
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
 
 import com.wb.payload.WeatherMapper;
 import com.wb.service.WeatherService;
 
-@RunWith(SpringRunner.class)
 @SpringBootTest
+@AutoConfigureMockMvc
 public class WeatherForecastApiApplicationUnitTests {
-    @Mock
-    private WeatherService weatherService;
-    @Mock
-    Future<WeatherMapper> futureWeather = new CompletableFuture<>();
-    @InjectMocks
-    WeatherForecastController controller = new WeatherForecastController(weatherService);
-    WeatherMapper weatherMapper;
 
-    @org.junit.jupiter.api.Test
-    public void contextLoads() {
-    }
+    private final WeatherService weatherService;
+    private final MockMvc mockMvc;
 
-    @Before
-    public void setUp1() throws Exception {
-        weatherMapper = new WeatherMapper();
-        when(futureWeather.isDone()).thenReturn(true);
-        when(futureWeather.get()).thenReturn(weatherMapper);
-        when(weatherService.fetchWeatherByCity(Mockito.anyString())).thenReturn(weatherMapper);
-        when(weatherService.testWeatherByCity(Mockito.anyString())).thenReturn(futureWeather);
+    @Autowired
+    public WeatherForecastApiApplicationUnitTests(WeatherService weatherService, MockMvc mockMvc) {
+        this.weatherService = weatherService;
+        this.mockMvc = mockMvc;
     }
 
     @Test
-    public void shouldReturnCityWeatherObject() throws InterruptedException, ExecutionException {
-        Future<WeatherMapper> fw = weatherService.testWeatherByCity("Warsaw");
-        while (!fw.isDone()) {
-            Thread.sleep(10);
-        }
-        assertEquals(weatherMapper, fw.get());
-    }
-
-    @Before
-    public void setUp2() throws Exception {
-        weatherMapper = new WeatherMapper();
-        when(futureWeather.isDone()).thenReturn(true);
-        when(futureWeather.get()).thenReturn(weatherMapper);
-        when(weatherService.fetchWeatherByCityAndTime(Mockito.anyString(), Mockito.anyLong(), Mockito.anyLong())).thenReturn(weatherMapper);
-        when(weatherService.testWeatherByCityAndTime(Mockito.anyString(), Mockito.anyLong(), Mockito.anyLong())).thenReturn(futureWeather);
+    public void apiFetchWeatherByCity() throws Exception {
+        this.mockMvc.perform(get("/api/v1/wb/city/{city}", "Warsaw,pl").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(matchAll(status().isOk()));
     }
 
     @Test
-    public void shouldReturnCityTimeWeatherObject() throws InterruptedException, ExecutionException {
-        Future<WeatherMapper> fw = weatherService.testWeatherByCityAndTime("Warsaw", 1581712620L, 1581799020L);
-        while (!fw.isDone()) {
-            Thread.sleep(10);
-        }
-        assertEquals(weatherMapper, fw.get());
+    public void serviceFetchWeatherByCity() throws Exception {
```

Comment:
test dobry procz tego że strzelasz do prawdziwego API. to o co prosiłem to zamockowanie odpowiedzi z Rest Template ktory jest w srodku tak zebys mogl przetestować różne scenariusze